Fix opencv crash

ckuethe opened the pull request:
This depends on the previous PR to add getopt. Now that I have getopt, it's easy to add extra flags to do useful things, like filtering improperly sized images to avoid opencv assertion failures.

EricClaeys commented:
For the --fontcolor in keogram.cpp, would it be possible to allow both "#00ff00" format and "000 255 000" format (enclosed in quotes)?  A lot of non-experienced users (and some experienced ones) find hex notation more difficult than a regular number.  For example, what's half of "ff", versus half of 255?
One way to implement would be if there's a space in the string, it's the second format, otherwise the first.

ckuethe commented:
I though about that but decided against it in the interests of simplicity of parsing. There are lots of html color calculators online.

EricClaeys commented:
@ckuethe I don't think making things easier for a programmer but harder for a user is the way to go.
By requiring the "000 255 000" be in quotes, the fontcolor will be treated as one argument and can be parsed like it's done in capture.cpp: `sscanf(argv[++i], "%d %d %d", &fontcolor[0], &fontcolor[1], &fontcolor[2]);`
The keogram.cpp code could either check for a space, or try the sscanf() and if it doesn't return 3, use the #ffffff method.

ckuethe commented:
Rebased.

ckuethe commented:
Also mitigates #492

ckuethe commented:
Just to clarify what's going in here, I found a few edge cases when reprocessing old images.
* number of channels unexpected (eg. trying to work with both color and monochrome images). fixed by changing colormap of offending image on the fly.
* image size varying (somehow I got unexpected sizes of images). fixed by adding skipping images with unexpected size
* keograms would be missing time markers (timestamps not read from disk correctly). fixed by adding a flag to extract timing information from the filename)

linuxkidd reviewed (approved):
Code looks sound!
